### PR TITLE
User/michaelshikh/2022 12 29

### PR DESCRIPTION
**Pull request by michaelshikh07**
## Description

- Added cmdlet `Get-AzKustoSku`
- Added parameter `CallerRole` for cmdlet `AzKustoDatabase` and `Update-AzKustoDatabase`
- Added support for new data connection kind `CosmosDb` for cmdlet `New-AzKustoDataConnection` and `Update-AzKustoDataConnection`
- Added parameters `DatabaseNameOverride` `DatabaseNamePrefix` `TableLevelSharingPropertyFunctionsToInclude` `TableLevelSharingPropertyFunctionsToExclude` for cmdlet `New-AzKustoAttachedDatabaseConfiguration`
## Checklist

- [ V] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Kusto/examples/Get-AzKustoSku.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+### Example 1: {{ Add title here }}
```
Comment (VeryEarly) [inline src/Kusto/examples/Get-AzKustoSku.md:1]: please add example for new cmdlet

**Comment by VeryEarly** [comment]
breaking changes are not allowed until major release, will hold this PR before it.

File: src/Kusto/examples/Get-AzKustoSku.md (hunk 1 above)
Comment (michaelshikh07) [reply src/Kusto/examples/Get-AzKustoSku.md]: added

**Comment by michaelshikh07** [comment]
> 
in the case of New-AzKustoDataConnection
<html>
<body>


The cmdlet 'New-AzKustoDataConnection' no longer supports the type 'Microsoft.Azure.PowerShell.Cmdlets.Kusto.Support.EventGridDataFormat' for parameter 'DataFormat'
--



</body>
</html>

KustoDataConnection is a case of polymorphism, there are 4 different data connections, "Kind" property is the type identifier.
DataFormat property was incorrectly declared as  EventGridDataFormat.
In fact we have 3 different types of dataFormat 
https://github.com/Azure/azure-rest-api-specs/blob/3fb19df9ee39171e411cbd28af0cf5ac743d143e/specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2022-12-29/kusto.json#L4495
https://github.com/Azure/azure-rest-api-specs/blob/3fb19df9ee39171e411cbd28af0cf5ac743d143e/specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2022-12-29/kusto.json#L4521
https://github.com/Azure/azure-rest-api-specs/blob/3fb19df9ee39171e411cbd28af0cf5ac743d143e/specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2022-12-29/kusto.json#L4547

In order to correctly declare and share this property across all data connections, the type was declared as a string which is the base type of the enum in this case. 
Beisi Zhou assisted on this type change

File: tools/StaticAnalysis/Exceptions/Az.Kusto/SignatureIssues.csv [hunk 2]
```diff
@@ -70,4 +70,7 @@
...
-"Az.Kusto","Update-AzKustoScript","Update-AzKustoScript","1","8410","Parameter ProxyUseDefaultCredentials of cmdlet Update-AzKustoScript does not follow the enforced naming convention of using a singular noun for a parameter name.","Consider using a singular noun for the parameter name."
\ No newline at end of file
+"Az.Kusto","Update-AzKustoScript","Update-AzKustoScript","1","8410","Parameter ProxyUseDefaultCredentials of cmdlet Update-AzKustoScript does not follow the enforced naming convention of using a singular noun for a parameter name.","Consider using a singular noun for the parameter name."
+"Az.Kusto","Invoke-AzKustoDataConnectionValidation","Invoke-AzKustoDataConnectionValidation","0","2020","The cmdlet 'Invoke-AzKustoDataConnectionValidation' no longer supports the type 'Microsoft.Azure.PowerShell.Cmdlets.Kusto.Support.EventGridDataFormat' for parameter 'DataFormat'.","Change the type for parameter 'DataFormat' back to 'Microsoft.Azure.PowerShell.Cmdlets.Kusto.Support.EventGridDataFormat'."
```
Comment (VeryEarly) [inline tools/StaticAnalysis/Exceptions/Az.Kusto/SignatureIssues.csv:74]: line 74 to line 76 are not signature issues, please move to breaking change issues

Comment (michaelshikh07) [reply tools/StaticAnalysis/Exceptions/Az.Kusto/SignatureIssues.csv:74]: I edited this by accident, and rolled it back to the main branch on this file

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**